### DISCUSSION TITLE
If you created your component with `nlp.create_pipe('name')`: remove nlp.create_pipe and call `nlp.add_pipe('name')` instead.

### DISCUSSION THREAD

**Discussion by TheContentGym** [explosion/spaCy#10732] [Help: Coding & Implementations] [answered]
**File:
==============**
import spacy
from pysbd.utils import PySBDFactory
nlp = spacy.blank('en')

nlp.add_pipe(PySBDFactory(nlp))
doc = nlp("It costs $5 to make a small python program. This is a demo to nacho.")
print(list(doc.sents))

Error
===========================
- If you created your component with `nlp.create_pipe('name')`: remove nlp.create_pipe and call `nlp.add_pipe('name')` instead.

- If you passed in a component like `TextCategorizer()`: call `nlp.add_pipe` with the string name instead, e.g. `nlp.add_pipe('textcat')`.

- If you're using a custom component: Add the decorator `@Language.component` (for function components) or `@Language.factory` (for class components / factories) to your custom component and assign it a name, e.g. `@Language.component('your_name')`. You can then run `nlp.add_pipe('your_name')` to add it to the pipeline.
=============
Question: what am I missing here?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @TheContentGym , 

The `nlp.add_pipe` function [takes in a factory name](https://spacy.io/api/language#add_pipe) (a string). If you're using a custom component, then I recommend looking [at this documentation on how to properly make one](https://spacy.io/usage/processing-pipelines#custom-components). Also, next time please format your code in Markdown (use [backticks](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code)) and paste the full traceback.
